tests/e2e/csi_snapshot_basic.go

```diff
@@ -3762,25 +3802,30 @@ var _ = ginkgo.Describe("Volume Snapshot Basic Test", func() {
 			gomega.Expect(err).NotTo(gomega.HaveOccurred())
 		}()
 
-		totalQuotaUsedAfter, _ = getTotalQuotaConsumedByStoragePolicy(ctx, restConfig,
-			storageclass.Name, namespace)
-		framework.Logf("totalquota_used_after_snapshotCreation :%v", totalQuotaUsedAfter)
-
 		ginkgo.By("Start hostd service on the host on which the PV is present")
 		startHostDOnHost(ctx, hostIP)
 
+		if isQuotaValidationSupported && supervisorCluster {
+			var expectedTotalStorage []string
+			expectedTotalStorage = append(expectedTotalStorage, diskSize, diskSize, snapshotSizeStr)
+
+			validateTotalQuota(ctx, restConfig, storageclass.Name, namespace, expectedTotalStorage,
+				totalQuotaUsedBefore, islatebinding)
+
+			validateQuotaUsageAfterResourceCreation(ctx, restConfig, storageclass.Name, namespace, pvcUsage,
```

Comment:
return proper error from this util.
If something fails in this util, let it return to parent call and should not fail in the util itself